pkg/monitor/cluster/podconditions.go

```diff
@@ -120,3 +123,33 @@ func (mon *Monitor) _emitPodContainerStatuses(ps *corev1.PodList) {
 		}
 	}
 }
+
+func (mon *Monitor) _emitPodContainerRestartCounter(ps *corev1.PodList) {
+	for _, p := range ps.Items {
+		if !namespace.IsOpenShiftSystemNamespace(p.Namespace) {
+			continue
+		}
+
+		for _, cs := range p.Status.ContainerStatuses {
+			if cs.RestartCount < restartCounterThreadshold {
+				continue
+			}
+
+			mon.emitGauge("pod.restartcounter", int64(cs.RestartCount), map[string]string{
+				"name":          p.Name,
```

Comment:
I'm a bit worried about cardinality of this dimension, but I don't know if there's a better way to distinguish different pod groups in the same namespace. Since we're already using that for other pod.* metrics it's fine here.

Just a note for further discussion.

Reply:
We can probably drop `nodeName` and/or `containerName` here. Namespace + podname should be enough to identify a problematic container/node.

Reply:
Happy to drop these fields.